src/common/models/data-source/data-source.ts

```diff
@@ -294,16 +303,19 @@ export class DataSource implements Instance<DataSourceValue, DataSourceJS> {
       dimensions,
       measures,
       timeAttribute,
-      defaultTimezone: parameters.defaultTimezone ? Timezone.fromJS(parameters.defaultTimezone) : DataSource.DEFAULT_TIMEZONE,
-      defaultFilter: parameters.defaultFilter ? Filter.fromJS(parameters.defaultFilter) : Filter.EMPTY,
-      defaultDuration: parameters.defaultDuration ? Duration.fromJS(parameters.defaultDuration) : DataSource.DEFAULT_DURATION,
+      defaultTimezone: parameters.defaultTimezone ? Timezone.fromJS(parameters.defaultTimezone) : null,
+      defaultFilter: parameters.defaultFilter ? Filter.fromJS(parameters.defaultFilter) : null,
+      defaultDuration: parameters.defaultDuration ? Duration.fromJS(parameters.defaultDuration) : null,
       defaultSortMeasure: parameters.defaultSortMeasure || (measures.size ? measures.first().name : null),
       defaultSelectedMeasures: parameters.defaultSelectedMeasures ? OrderedSet(parameters.defaultSelectedMeasures) : null,
       defaultPinnedDimensions: parameters.defaultPinnedDimensions ? OrderedSet(parameters.defaultPinnedDimensions) : null,
       refreshRule,
       maxTime
     };
-    if (external) value.external = external;
+    if (cluster) {
+      if (engine !== cluster.name) throw new Error(`Engine '${engine}' was given but '${cluster.name}' cluster was supplied (must match)`);
```

Comment:
one day `dataSource.engine` will be renamed to `dataSource.cluster` and it corresponds to the cluster name which could be `"druid-prod"` or `"metrics-internal"` or whatever.